lib/chef/knife/bootstrap/client_builder.rb

```diff
@@ -80,9 +80,12 @@ def node_name
           knife_config[:chef_node_name]
         end
 
-        # @return [String] enviroment from the knife_config
+        # @return [String] enviroment from the knife_config or first_boot_attributes, if none specified by knife_config
         def environment
-          knife_config[:environment]
+          # prefer environment from cli over environment from first-boot attributes
+          knife_config[:environment] or
+            (first_boot_attributes and first_boot_attributes[:environment]) or
+            chef_config[:environment]
```

Comment:
@btm Yeah, that's generally correct. It's captured [here](https://github.com/chef/chef/blob/mcquin/first-boot-environment/lib/chef/knife/bootstrap/client_builder.rb#L30-L33).